tool/docker/devel/native/ubuntu/cuda/py3/Dockerfile

```diff
@@ -20,34 +20,51 @@ FROM nvidia/cuda:9.0-cudnn7-devel-ubuntu16.04
 
 # install dependencies
 RUN apt-get update \
-    && apt-get install -y --no-install-recommends git build-essential autoconf libtool cmake libprotobuf-dev libopenblas-dev libpcre3-dev protobuf-compiler wget openssh-server swig python3-dev python3-pip python3-setuptools\
-    && apt-get clean && apt-get autoremove && apt-get autoclean \
+    && apt-get install -y --no-install-recommends \
+        git \
+        build-essential \
+        autoconf \
+        libtool \
+        cmake \
+        libprotobuf-dev \
+        libopenblas-dev \
+        libpcre3-dev \
+        protobuf-compiler \
+        wget \
+        openssh-server \
+        swig \
+        python3-dev \
+        python3-pip \
+        python3-setuptools \
+    && apt-get clean \
+    && apt-get autoremove \
+    && apt-get autoclean \
     && rm -rf /var/lib/apt/lists/* \
-    && pip3 install -U wheel numpy setuptools unittest-xml-reporting protobuf future
-
-
-# install swig > 3.0.10 for ubuntu < 16.04
-# RUN wget http://prdownloads.sourceforge.net/swig/swig-3.0.10.tar.gz && \
-#     tar zxf swig-3.0.10.tar.gz && cd swig-3.0.10 && \
-#     ./configure && make && make install
-
-# set environment
-
-# ENV CMAKE_INCLUDE_PATH /usr/local/cuda/include:${CMAKE_INCLUDE_PATH}
-# ENV CMAKE_LIBRARY_PATH /usr/local/cuda/lib64:${CMAKE_LIBRARY_PATH}
-
-# config ssh service
-RUN mkdir /var/run/sshd
-RUN echo 'root:singa' | chpasswd
-# for ubuntu 14.04
-# RUN sed -i 's/PermitRootLogin without-password/PermitRootLogin yes/' /etc/ssh/sshd_config
-# for ubuntu 16.04 prohibit
-RUN sed -i 's/PermitRootLogin prohibit-password/PermitRootLogin yes/' /etc/ssh/sshd_config
-# SSH login fix. Otherwise user is kicked off after login
-RUN sed 's@session\s*required\s*pam_loginuid.so@session optional pam_loginuid.so@g' -i /etc/pam.d/sshd
-
-# dump environment variables into files, so that ssh can see also
-RUN env | grep _ >> /etc/environment
+    && pip3 install -U --no-cache \
+        wheel \
+        numpy \
+        setuptools \
+        unittest-xml-reporting \
+        protobuf \
+        future \
+    # install swig > 3.0.10 for ubuntu < 16.04
+    # RUN wget http://prdownloads.sourceforge.net/swig/swig-3.0.10.tar.gz && \
+    #     tar zxf swig-3.0.10.tar.gz && cd swig-3.0.10 && \
+    #     ./configure && make && make install
```

Comment:
Since this dockerfile uses ubuntu16.04, it must install swig-3.0.10 because it doesn't come by default. So these three lines should not be commented.

Also installing swig from source like this will put the swig executable in  /usr/local/bin/swig, while cmake will find the old one in /usr/bin/swig3.0 . So we may add this line after the "make install":

cp  /usr/local/bin/swig /usr/bin/swig3.0

Reply:
+1